### PR TITLE
ci: store .snap files as build artifacts

### LINKED ISSUE #740: Store artifacts for PRs in CircleCI
I think it would be helpful to store the `.snap` build artifacts for PRs, so people can easily download and test them without having to build the snap on their own.

I'm not sure if there is a storage limit in CircleCI (They didn't write anything about that in [their documentation](https://circleci.com/docs/2.0/artifacts/)).

What do you think about this, @kyrofa (or anyone else)?

### PR DESCRIPTION
**Pull request by bn4t**
Fixes #740

### CODE REVIEW

**Comment by bn4t** [comment]
Going to close this PR and open a new one for this. Somehow newly added commits don't show up in this PR.

**Comment by bn4t** [comment]
Going to close this PR and open a new one for this. Somehow newly added commits don't show up in this PR.

**Comment by kyrofa** [comment]
Yeah github has been having issues since yesterday, I haven't been able to do much, either :'( .

**Comment by bn4t** [comment]
This seems to be working now. 
I'm not sure how access to the artifacts exactly works: If I'm logged in I'm able to access the artifacts, however when I'm logged out I'm unable to access them. 
Ideally the artifacts should be accessible by everyone.

Would it be possible to configure this in the CircleCI settings or is this just something we'd have to live with? @kyrofa

**Comment by bn4t** [comment]
Haven't seen it before, but it looks great. Pretty much what we need. 
If you're ok with it, I'll add it to the CI script.

In the future [this PR](https://github.com/themadcreator/circle-github-bot/pull/16) might be of relevance (if it isn't already merged by then).

**Comment by kyrofa** [comment]
Interesting, the top of the readme says "Now compatible with Circle 2.0".

**Comment by bn4t** [comment]
The readme is technically right, the script just uses a deprecated environment variable. However we're good as long CircleCi doesn't remove this variable (or the PR gets merged).

**Comment by bn4t** [comment]
Hmm, the CI build keeps failing because of a weird error (which seems unrelated to recent changes):
`E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/l/linux/linux-libc-dev_4.4.0-137.163_amd64.deb  404  Not Found [IP: 91.189.88.152 80]`

The same error happened in previous builds and then vanished again. 
No clue why this is happening. Do you have any idea? @kyrofa

**Comment by kyrofa** [comment]
A mistake on my part. Wait for [this to land](https://github.com/nextcloud/nextcloud-snap/pull/763/files#diff-1d37e48f9ceff6d8030570cd36286a61R29) and all should be right with the world.

**Comment by bn4t** [comment]
No clue on how to fix the error that's currently happening... It occures on [this line](https://github.com/nextcloud/nextcloud-snap/pull/742/files#diff-5a58ca028f9bc131185ab6b601f0b4bcR4).
Do any of you maybe have an idea what the problem could be? @kyrofa @pachulo @r4co0n

File: .circleci/config.yml [hunk 1]
```diff
@@ -42,10 +42,23 @@ jobs:
...
 
+     - run:
+         # Copy artifacts, install nodejs and publish download link in PR
+         command: |
+           sudo mkdir /artifacts/
+           sudo cp *.snap /artifacts/nextcloud.snap
+           wget -qO- https://deb.nodesource.com/setup_10.x | sudo -E bash -
+           sudo apt-get install -y nodejs
+           npm install circle-github-bot
+           sh $(pwd)/.circleci/publish-artifact.js
```
Comment (kyrofa) [inline .circleci/config.yml]: You probably want to either execute this with node, or just execute it directly and let the shebang take effect.

Comment (bn4t) [reply .circleci/config.yml]: Thanks for the tip. I thought the shebang would be applied despite using `sh`.

**Comment by kyrofa** [comment]
@bn4t can I safely assume that the bot will make the comment under the name of that account? i.e. should I create a bot account, then?

**Comment by kyrofa** [comment]
Alright @bn4t, re-running now.

**Comment by kyrofa** [comment]
Looks like that's a 404.

**Comment by bn4t** [comment]
Updated the artifact path. Hopefully fixes the `404` error.

**Comment by kyrofa** [comment]
Hmm, I still can't download this.

**Comment by bn4t** [comment]
Yeah, I'll need to rewrite the bot. I'll do that as soon as I find the time.

**Comment by bn4t** [comment]
Thanks @pachulo.
What do you think about the text the bot writes? Should I change something? @pachulo @kyrofa

**Comment by bn4t** [comment]
Is this ready to merge yet, or are there still things to be done? @kyrofa

**Comment by kyrofa** [comment]
@bn4t this PR inspired https://github.com/kyrofa/github-snap-builder, and the Nextcloud snap will be the pilot project for it. Every pull request should have a snap built and automatically published into `beta/pr-#` like I've been doing manually. Only for amd64 still, but that's still an improvement. Anyway, as a result, this PR isn't necessary for now.

**Pull request closed without merging**